mlrun/datastore/base.py

```diff
@@ -217,6 +234,7 @@ def reader(*args, **kwargs):
                     ):
                         raise ex
 
+                    #  TODO fix this code, if end_time has tz and start_time is None?
```

Comment:
Looks like you're right. Please add a test to confirm the issue, and fix it. We're not just going to leave a TODO in the code about an obvious bug.

Reply:
I started to do so, but it seems to be more complicated than it looks. I will open an issue and add the right issue number to the TODO.